extensions/binding/org.eclipse.smarthome.binding.coapconnectedlighting/src/main/java/org/eclipse/smarthome/binding/coapconnectedlighting/handler/CoAPConnectedLightingHandler.java

```diff
@@ -0,0 +1,361 @@
+        if (message.compareTo("ON") == 0) {
+            coapLightSwitch.append(",1,pp,100,");
+        } else if (message.compareTo("OFF") == 0) {
+            coapLightSwitch.append(",1,pp,0,");
+        }
+        CoapResponse clientResponse = coapClient.put(coapLightSwitch.toString(), TEXT_PLAIN);
+
+        logger.debug(clientResponse.advanced().getSource().toString());
+    }
+
+    private void putlightDimmer(String message) {
+        coapClient.setURI(
+                "coap://" + COAP_SERVER_IP_ADDRESS + ":" + COAP_SERVER_PORT + "/InternetOfTiny/actuators/actuator1");
+        StringBuilder coapLightSwitch = new StringBuilder();
+        coapLightSwitch.append(",1,pp," + message + ",");
+
+        CoapResponse clientResponse = coapClient.put(coapLightSwitch.toString(), TEXT_PLAIN);
+
+        logger.debug(clientResponse.advanced().getSource().toString());
+    }
+
+    private void startBackgroundTemperatureRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    getTemprature();
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshTemperatureJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
+    }
+
+    private void startBackgroundHumidityRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    getHumidity();
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshHumidityJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
+    }
+
+    private void startBackgroundPressureRefresh() {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    getPressure();
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshPressureJob = scheduler.scheduleAtFixedRate(runnable, 0, refresh.intValue(), TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (channelUID.getId().equals(CHANNEL_DimmableLight)) {
+                if (command instanceof PercentType) {
+                    PercentType commandString = ((PercentType) command);
+                    putlightDimmer(commandString.toString());
+                } else if (command instanceof DecimalType) {
+                    IncreaseDecreaseType commandString = ((IncreaseDecreaseType) command);
```

Comment:
despite the command being of type `DecimalType` you cast to `IncreaseDecreaseType ` here.

Reply:
I am going to defer resolution of this typecast to a later stage. I will need to do some debugging for this one.